cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamSpecification.java

```diff
@@ -17,15 +17,19 @@
 package co.cask.cdap.api.data.stream;
 
 import java.util.Objects;
+import javax.annotation.Nullable;
 
 /**
  * Specification for {@link Stream}.
  */
 public final class StreamSpecification {
   private final String name;
+  @Nullable
+  private final String description;
 
-  private StreamSpecification(final String name) {
+  private StreamSpecification(String name, String description) {
```

Comment:
Defaulting should not happen in the core. The core and the APIs should return null. Instead, should be added at the presentation layer.

Here are what need to be added at presentation layer. 
- **Stream** : No description provided for Stream. 
- **Dataset** : No description provided for this Dataset. 
- **Stream View** : No description provided for this Stream View.